src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -840,6 +869,19 @@ FunctionResultContent CreateFunctionResultContent(FunctionInvocationResult resul
         return result;
     }
 
+    /// <summary>This method will invoke the fuction within the try block.</summary>
+    /// <param name="context">The function invocation context.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
+    /// <returns>The function invocation context and result.</returns>
+    protected virtual async Task<(FunctionInvocationContext context, object? result)> TryInvokeFunctionAsync(FunctionInvocationContext context, CancellationToken cancellationToken)
```

Comment:
It would be odd for both `InvokeFunctionAsync` and `TryInvokeFunctionAsync` to become part of the public API surface, as it's very unclear (from the naming) how they differ.

Reply:
Maybe the right design is for us to make `InvokeFunctionAsync` *not* be protected/virtual and have only the inner thing (TryInvokeFunctionAsync as you've called it) be protected/virtual.

The other thing just adds logging (and sets context), so we could make that private and call it `LogAndInvokeFunction` or something, then the new method you've added could have the nicer name `InvokeFunctionAsync` and be protected and virtual.

I think what you've added here is a better thing to be protected/virtual since if someone overrides this, they likely *only* want to change the actual invocation and not change anything about logging.

Reply:
> Maybe the right design is for us to make InvokeFunctionAsync not be protected/virtual and have only the inner thing (TryInvokeFunctionAsync as you've called it) be protected/virtual.

I agree here, this was the original method and I didn't want to introduce breaking changes, but, yes, if we can have the current `InvokeFunctionAsync` internalized and the `Try` become the new `InvokeFunctionAsync`, thats correct, in our specialized client we didn't need to override this (InvokeFunctionAsync).